### PR TITLE
feat: add zcap/ocap request capability to ethereum auth provider

### PR DESCRIPTION
**Pull request by haardikk21**
First pass at solving https://github.com/ceramicnetwork/js-ceramic/issues/1713

### CODE REVIEW

**Comment by ukstv** [comment]
@haardikk21 Any news on the changes?

File: packages/blockchain-utils-linking/src/__tests__/ethereum.test.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import * as sigUtils from 'eth-sig-util'
 import { ContractFactory, Contract } from '@ethersproject/contracts'
 import { encodeRpcMessage } from '../util'
 import * as ethereum from '../ethereum'
+import { OcapParams, OcapTypes } from '..'
```
[the remark this answers is not in the document]
Comment (haardikk21) [reply packages/blockchain-utils-linking/src/__tests__/ethereum.test.ts:7]: Oops, I think this was a VSCode auto import, will fix

File: packages/blockchain-utils-linking/src/ethereum.ts [hunk 2]
```diff
@@ -47,6 +56,25 @@ export class EthereumAuthProvider implements AuthProvider {
     return createLink(did, accountId, this.provider, this.opts)
   }
 
+  async requestCapability(params: OcapParams): Promise<Ocap> {
+    console.warn(
+      'WARN: requestCapability os an experimental API, that is subject to change any time.'
+    )
+    const account = await this.accountId()
+    const requestMessage = getOcapRequestMessage({
```
[the remark this answers is not in the document]
Comment (haardikk21) [reply packages/blockchain-utils-linking/src/ethereum.ts:64]: makes sense

**Comment by haardikk21** [comment]
> @haardikk21 Any news on the changes?

addressed

**Review by ukstv — APPROVED** [review]

**Pull request merged**